webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js

```diff
@@ -2,19 +2,36 @@ import React from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
+import { useDispatch } from 'react-redux';
+
+import { setSettingEditing } from '../../SettingRecords/SettingRecordsActions';
+import useSettingModal from '../../SettingUpdateModal/useSettingModal';
 
 import { valueToString, hasDefault, inStrong } from '../SettingsTableHelpers';
 
 const SettingCellInner = props => {
-  const { setting, className, onEditClick, ...rest } = props;
+  const { setting, className, ...rest } = props;
 
   const cssClasses = classNames(className, {
     'editable-empty': !setting.value && setting.settingsType !== 'boolean',
     'masked-input': setting.encrypted,
   });
 
+  if (!setting.readonly){
+    const { setModalOpen } = useSettingModal();
+    const dispatch = useDispatch();
+    const onEditClick = async setting => {
+      await dispatch(setSettingEditing(setting));
```

Comment:
It isn't related, but `setSettingEditing` isn't async, I wonder why `onEditClick` needs to be asynchronous.

Reply:
looks like it doesn't, fixed